apiary.apib

```diff
@@ -0,0 +1,347 @@
+FORMAT: 1A
+HOST: http://deploy.cwp.govt.nz
+
+# Deploynaut API
+
+This API is a newly developed interface for the CWP Dashboard. Data structure is based on the
+[JSON API specifications](http://jsonapi.org/).
+
+
+# Using the API [/naut]
+
+In order to use the API, you will need to acquire a token via Service Desk request. In order to authenticate,
+you should base-64 encode your email/token combination e.g.
+
+```
+Email => sherbert.lemon@hogwarts.uk
+Token => fa88787fcd99787
+Pre-encode string => sherbert.lemon@hogwarts.uk:fa88787fcd99787
+Authorization token => c2hlcmJlcnQubGVtb25AaG9nd2FydHMudWs6ZmE4ODc4N2ZjZDk5Nzg3
+```
+
+Any request you make to an endpoint should include the following:
+
+## Making a request: Headers [GET]
+
+These headers will authenticate you against Deploynaut to evaluate whether you have the permissions necessary
+to access the information you are requesting. You can send a GET request to deploy.cwp.govt.nz/naut/profile
+to test that you are authenticating correctly - it will return your user information.
+
++ Request (application/json)
+
+    + Headers
+
+            X-api-version: 1.0
+            Authorization: Basic c2hlcmJlcnQubGVtb25AaG9nd2FydHMudWs6ZmE4ODc4N2ZjZDk5Nzg3
+            Accept: application/json
+            Host: deploy.cwp.govt.nz
+
+
++ Response 200 (application/json)
+
+        {
+            "data": [
+                {
+                    "type": "users",
+                    "id": "2379",
+                    "name": "Albus Dumbledore",
+                    "email": "sherbert.lemon@hogwarts.uk"
+                }
+            ]
+        }
+
+
+
+# Stacks [/naut/projects]
+
+## List all stacks [GET /naut/projects]
+
+Will return a list of all stacks that you have the authority to view, with relevant datapoints.
+
++ Response 200 (application/json)
+
+        {
+            "count": 1,
+            "data": [
+                {
+                    "type": "stacks",
+                    "name": "CWP001",
+                    "title": "CWP Stack 001",
+                    "created": "02-07-2014",
+                    "environments": {
+                        "count": 2,
+                        "data": [
+                            {
+                                "type": "environments",
+                                "name": "prod",
+                                "build_sha": "d0bcf80381579a529957d77a6b0a69303afebe7e",
+                                "deployer": {
```

Comment:
> Also this is a bit weird in that a user could belong to multiple teams. Would it then make sense to have a /naut/users/$id

It might make sense in the future, but for MVP, I'd not bother creating /naut/users/$id endpoint, because you can get this information from /naut/project/$name/team already.

I'd remove drop "deployers" key in both project and environment API calls completely, because how do you even know that people will want to know deployers, but not release managers?